File: CNC modeling 2.0/src/main/java/com/sergey/pisarev/controller/RenameFrameNumbersController.java
package com.sergey.pisarev.controller;

import com.sergey.pisarev.model.RenameFrameNumbers;
import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.scene.control.TextField;
import javafx.stage.Stage;

public class RenameFrameNumbersController {

    public static Stage STAGE;

    @FXML
    TextField textFieldFirstNumber=new TextField();

    @FXML
    TextField textFieldStep=new TextField();

    @FXML
    public void buttonOnClickOk(ActionEvent actionEvent) {
        if(!textFieldFirstNumber.getText().isEmpty()&&!textFieldStep.getText().isEmpty()&&!MainController.mainController.getTextCodeArea().isEmpty()){
            MainController.mainController.setTextCodeArea(RenameFrameNumbers.rename(MainController.mainController.getTextCodeArea(),Integer.parseInt(textFieldFirstNumber.getText()),Integer.parseInt(textFieldStep.getText())));
            STAGE.close();
        }
    }

    @FXML
    public void buttonOnClickCancel(ActionEvent actionEvent) {
        STAGE.close();
    }
}

File: CNC modeling 2.0/src/main/java/com/sergey/pisarev/controller/RenameFrameNumbersController.kt
package com.sergey.pisarev.controller

import com.sergey.pisarev.model.RenameFrameNumbers.rename
import javafx.event.ActionEvent
import javafx.fxml.FXML
import javafx.scene.control.TextField
import javafx.stage.Stage

class RenameFrameNumbersController {
    @FXML
    var textFieldFirstNumber = TextField()

    @FXML
    var textFieldStep = TextField()
    @FXML
    fun buttonOnClickOk(actionEvent: ActionEvent?) {
        if (textFieldFirstNumber.text.isNotEmpty() && textFieldStep.text.isNotEmpty() && MainController.mainController!!.textCodeArea!!.isNotEmpty()) {
            MainController.mainController!!.textCodeArea = rename(MainController.mainController?.textCodeArea, textFieldFirstNumber.text.toInt(), textFieldStep.text.toInt())
            STAGE!!.close()
        }
    }

    @FXML
    fun buttonOnClickCancel(actionEvent: ActionEvent?) {
        STAGE!!.close()
    }

    companion object {
        @JvmField
        var STAGE: Stage? = null
    }
}